### DISCUSSION TITLE
Image optimization on the fly

### DISCUSSION THREAD

**Discussion by dodyg** [SixLabors/ImageSharp.Web#120] [Q&A] [answered]
Is it possible to perform casual web image optimization for png and jpeg (such as https://tinypng.com/, https://tinyjpg.com/) on top of the image resizer on ImageSharp.Web?

**Comment by JimBobSquarePants** [comment, marked as answer]
Hi @dodyg 

We do yes but I need to ship a fix before it's particularly useful as we're sending the wrong (pre action) content length to the browser.

There's a fix in the PR here
https://github.com/SixLabors/ImageSharp.Web/pull/121/files#diff-b4c81dd77b1c1eab3717c630947e2e4aR315

Once that's in you'll be able to use the event `OnProcessedAsync` which provides you the process stream to further optimize.
